h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -216,8 +215,21 @@ private Frame prepareLevelOneFrame(String levelOneKey, Model[] baseModels, Frame
       if (baseModels.length != baseModelPredictions.length)
         throw new H2OIllegalArgumentException("Base models and prediction arrays are different lengths.");
 
-      if (null == levelOneKey) levelOneKey = "levelone_" + _model._key.toString();
-      Frame levelOneFrame = new Frame(Key.<Frame>make(levelOneKey));
+      if (null == levelOneKey) levelOneKey = "levelone_" + _model._key.toString() + "_" + _parms._metalearner_transform.toString();
+
+      // TODO: what if we're running multiple in parallel and have a name collision?
+      Frame old = DKV.getGet(levelOneKey);
+      if (old != null && old instanceof Frame) {
+        Frame oldFrame = (Frame) old;
+        // Remove ALL the columns so we don't delete them in remove_impl.  Their
+        // lifetime is controlled by their model.
+        oldFrame.removeAll();
+        oldFrame.write_lock(_job);
```

Comment:
you need to get a write-lock prior to changing a frame, otherwise, you might be changing an instance that is already locked

Reply:
Thank you! I was wondering about that - this code was just moved couple lines up. This was there for the last 4 years so I was assumed it was correct.